### PR TITLE
Proposed changes

### PR DESCRIPTION
**Pull request by dg0yt**
You may pick parts as you like. I didn't test more than `vcpkg install` for x64-linux and x64-linux-dynamic.

- `debug/include` is never written.
- vcpkg wants to have all tools to be in `tools`, even on linux. [I don't like it, but it is a fact.](https://github.com/microsoft/vcpkg/issues/17607)
- The CMake config may just assumed that everything is installed correctly. Otherwise it is a port bug, not a "package not found".
- The cmake config file uses the package name pattern for case-insensitive matches. CMake will set `<requested-name>_FOUND`. I dropped explicit setting of `directx-dxc_FOUND`.
- Just generic target properties.
- Directly set `DIRECTX_DXC_TOOL` as a cache variable. Doesn't hurt, and user might override it to a host path for cross builds.
- Unchanged: The variable name pattern of `DIRECTX_DXC_TOOL` neither matches the former explicit `directx-dxc_FOUND` nor the general `<requested-name>_FOUND` (unless the user asked for `find_package(DIRECTX_DXC)`).
- On linux, the libs use `libz.so.1`. With port `zlib` installed, this will resolve to that port's binary instead of a system binary. To avoid subtle effects from installation order, I suggest to make zlib an explicit dependency at least for dynamic triplets.

### CODE REVIEW

File: ports/directx-dxc/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,7 @@
-set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
```
Comment (walbourn) [inline ports/directx-dxc/portfile.cmake:1]: Why did you remove this? Is this no longer a policy?

File: ports/directx-dxc/directx-dxc-config.cmake.in [hunk 2]
```diff
@@ -1,44 +1,24 @@
...
-    set(directx-dxc_FOUND FALSE)
-
-endif()
+set(DIRECTX_DXC_TOOL "${_dxc_root}/@tool_path@" CACHE PATH "Location of the dxc tool")
+mark_as_advanced(DIRECTX_DXC_TOOL)
+
+add_library(Microsoft::DirectXShaderCompiler SHARED IMPORTED)
+set_target_properties(Microsoft::DirectXShaderCompiler PROPERTIES
+   IMPORTED_LOCATION                    "${_dxc_root}/@dll_dir@/@dll_name_dxc@"
+   IMPORTED_IMPLIB                      "${_dxc_root}/lib/@lib_name@"
```
Comment (walbourn) [inline ports/directx-dxc/directx-dxc-config.cmake.in:10]: When I try to build for `RelWithDebInfo` it fails. Any idea why?

I'm using `set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO Release)`

File: ports/directx-dxc/portfile.cmake (hunk 1 above)
Comment (walbourn) [reply ports/directx-dxc/portfile.cmake:1]: In vcpkg-tools on GitHub it looks like this policy exists but is never referenced. I take it is deprecated?

File: ports/directx-dxc/directx-dxc-config.cmake.in (hunk 2 above)
Comment (walbourn) [reply ports/directx-dxc/directx-dxc-config.cmake.in:10]: So this is why I went with the DEBUG and RELEASE version before... using the non-config version fails for RelWithDebInfo for some reason.

Comment (walbourn) [reply ports/directx-dxc/directx-dxc-config.cmake.in:10]: I have to use `CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO` for other ports to get RelWithDebInfoto map to Release.

The problem is that the new target doesn't work anymore with that scenario.  It looks like the fix is to add:

```
   MAP_IMPORTED_CONFIG_MINSIZEREL       ""
   MAP_IMPORTED_CONFIG_RELWITHDEBINFO   ""
```

Comment (dg0yt) [reply ports/directx-dxc/directx-dxc-config.cmake.in:10]: https://github.com/microsoft/vcpkg/pull/37485#issuecomment-2001789218

File: ports/directx-dxc/portfile.cmake [hunk 3]
```diff
@@ -1,11 +1,7 @@
...
-
 set(DIRECTX_DXC_TAG v1.8.2403)
 set(DIRECTX_DXC_VERSION 2024_03_07)
 
-if (NOT VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
-   message(STATUS "Note: ${PORT} always requires dynamic library linkage at runtime.")
-endif()
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```
Comment (dg0yt) [inline ports/directx-dxc/portfile.cmake:4]: Maybe I didn't fully pay attention to why it was done the original way.
Looking at past PRs, I realize that you deliberately put it the other way because it is independent of vcpkg CRT linkage on Windows and should be blocked on x64-windows-static. `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` requires the user to customize per-port  library linkage in static triplets to overcome this check.
OTOH the linux libs are dynamically linked to `libc.so.6`, i.e. using dynamic CRT linkage.

Comment (walbourn) [reply ports/directx-dxc/portfile.cmake:4]: Put it back then. Thanks!
